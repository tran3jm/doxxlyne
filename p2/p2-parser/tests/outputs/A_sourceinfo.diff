--- outputs/A_sourceinfo.txt	2022-10-06 22:48:27.919800601 -0400
+++ expected/A_sourceinfo.txt	2022-08-19 09:55:11.000000000 -0400
@@ -1,2 +1,11 @@
-a
-+
\ No newline at end of file
+Program [line 1]
+  FuncDecl name="main" return_type=int parameters={} [line 1]
+    Block [line 2]
+      VarDecl name="a" type=int is_array=no array_length=1 [line 3]
+      Assignment [line 4]
+        Location name="a" [line 4]
+        Binaryop op="+" [line 4]
+          Literal type=int value=4 [line 4]
+          Literal type=int value=5 [line 4]
+      Return [line 5]
+        Location name="a" [line 5]
