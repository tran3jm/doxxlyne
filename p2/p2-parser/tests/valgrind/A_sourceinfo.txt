==2213613== Memcheck, a memory error detector
==2213613== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2213613== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==2213613== Command: ../decaf inputs/add.decaf
==2213613== 
Program [line 1]
==2213613== 
==2213613== HEAP SUMMARY:
==2213613==     in use at exit: 0 bytes in 0 blocks
==2213613==   total heap usage: 1,710 allocs, 1,710 frees, 200,712 bytes allocated
==2213613== 
==2213613== All heap blocks were freed -- no leaks are possible
==2213613== 
==2213613== For lists of detected and suppressed errors, rerun with: -s
==2213613== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
