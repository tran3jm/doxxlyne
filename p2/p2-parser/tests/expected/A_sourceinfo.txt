Program [line 1]
  FuncDecl name="main" return_type=int parameters={} [line 1]
    Block [line 2]
      VarDecl name="a" type=int is_array=no array_length=1 [line 3]
      Assignment [line 4]
        Location name="a" [line 4]
        Binaryop op="+" [line 4]
          Literal type=int value=4 [line 4]
          Literal type=int value=5 [line 4]
      Return [line 5]
        Location name="a" [line 5]
